### PR TITLE
Buildah: drop base image results

### PR DESCRIPTION
**Pull request by chmeliik**
[STONEBLD-2326](https://issues.redhat.com//browse/STONEBLD-2326)

Add version 0.2 of buildah, buildah-oci-ta, buildah-remote, buildah-remote-oci-ta

The changes that one actually cares about are 
* `buildah 0.2: drop base images results`
* `buildah 0.2: remove unused params`
* `pipelines: update to buildah 0.2`

The rest is adjusting our ~two~ three task-generation mechanisms to make them generate all the other buildah variants

# Before you complete this pull request ...

Look for any open pull requests in the repository with the title "e2e-tests update" and 
see if there are recent e2e-tests updates that will be applicable to your change.

### CODE REVIEW

**Comment by chmeliik** [comment]
Won't be done in this PR, may be done in a separate PR:
* re-generate pipelines READMEs (far too many unrelated changes)
* remove BASE_IMAGES_DIGESTS from non-buildah tasks
  * rpm-ostree: will remove in a separate PR (the result is entirely useless)
  * s2i-*: the tasks are neglected, probably unused. Will leave them alone or remove the whole tasks.
* remove BASE_IMAGES_DIGESTS from buildah-rhtap
  * it doesn't add the base images to the SBOM, so the to-be-dropped result has no replacement
  * on the other hand, the docker-build-rhtap sample pipeline doesn't use them at all, so we can drop the result while we have the chance

**Comment by chmeliik** [comment]
> result "BASE_IMAGES_DIGESTS" not found in TaskRuns of PipelineRun build-templates-e2e/test-comp-vboo-s8p5j

gotta update e2e tests as well

**Comment by chmeliik** [comment]
re-based on main, hopefully re-applied changes correctly

File: hack/generate-buildah-remote.sh
Comment (zregvart) [inline hack/generate-buildah-remote.sh:12]: Nitpick, this might be safer, and I like the curlies, they make it a bit more readable:

```suggestion
    /tmp/remote-generator --buildah-task="${SCRIPTDIR}/../task/buildah/${version}"/buildah.yaml \
        --remote-task="${SCRIPTDIR}/../task/buildah-remote/${version}/buildah-remote.yaml"
    /tmp/remote-generator --buildah-task="${SCRIPTDIR}/../task/buildah-oci-ta/${version}/buildah-oci-ta.yaml" \
        --remote-task="${SCRIPTDIR}/../task/buildah-remote-oci-ta/${version}/buildah-remote-oci-ta.yaml"
```

**Review by zregvart — APPROVED** [review]
LGTM

File: task/buildah/0.2/buildah.yaml
Comment (chmeliik) [inline task/buildah/0.2/buildah.yaml:37]: @arewm had a good observation in Slack: we wanted to remove these params in the next version. I'll do that here as well

Comment (chmeliik) [reply task/buildah/0.2/buildah.yaml]: Removed them

File: hack/generate-buildah-remote.sh
Comment (chmeliik) [reply hack/generate-buildah-remote.sh]: Fixed

**Comment by chmeliik** [comment]
Dropped the temp commit. Tests will fail until https://github.com/konflux-ci/e2e-tests/pull/1265 is merged and until we merge the results e2e-tests-update PR

**Review by arewm — APPROVED** [review]

**Review by tkdchen — APPROVED** [review]

**Comment by chmeliik** [comment]
Reminder to self: don't merge this, need to re-apply buildah/0.1 changes that are about to be merged to main

**Comment by chmeliik** [comment]
Rebased on main, re-applied changes

**Pull request merged**
